test/data/mock-state.ts

```diff
@@ -0,0 +1,1462 @@
+
+const buildNftMock = (index: number): Nft => {
+  return {
+    address: `0x${index}`,
+    tokenId: `tokenId${index}`,
+    name: `NFT ${index}`,
+    image: `test${index}.jpg`,
+    description: `Description ${index}`,
+    standard: `ERC1155`,
+  };
+};
+
+const getRandomPastDate = () => {
+  const now = new Date();
+  // Random date within the last 90 days
+  const randomDaysAgo = Math.floor(Math.random() * 90);
+  const randomHours = Math.floor(Math.random() * 24);
+  const randomMinutes = Math.floor(Math.random() * 60);
+
+  const pastDate = new Date(now);
+  pastDate.setDate(now.getDate() - randomDaysAgo);
+  pastDate.setHours(randomHours);
+  pastDate.setMinutes(randomMinutes);
+
+  return pastDate.toISOString();
+};
+
+const buildRandomNotification = (
+  index: number,
+): NotificationServicesController.INotification => {
+  return {
+    id: `notification-${index}`,
+    type: TRIGGER_TYPES.FEATURES_ANNOUNCEMENT,
+    createdAt: getRandomPastDate(),
+    isRead: false,
+    data: {
+      title: `Mock Notification ${index}`,
+      longDescription: `<p>This is a mock notification description ${index}. It contains sample text to demonstrate how notifications appear in the system.</p>`,
+      image: {
+        url: `https://example.com/image-${index}.png`,
+        title: `Feature Announcement ${index}`,
+      },
+    },
+  };
+};
+
+const SIGNATURE_REQUEST_MOCK: SignatureRequest = {
+  chainId: SEPOLIA_TESTNET,
+  id: '123-456',
+  messageParams: {
+    data: '0xABC123',
+    from: ACCOUNT_THREE_ADDRESS,
+    origin: UNISWAP_ORIGIN,
+  },
+  networkClientId: SEPOLIA_TESTNET_NETWORK_CLIENT_ID,
+  status: SignatureRequestStatus.Signed,
+  time: Date.now(),
+  type: SignatureRequestType.PersonalSign,
+};
+
+const MOCK_TOKEN_DATA: TokenListToken = {
+  address: '0x0000000000000000000000000000000000000000',
+  name: 'Ether',
+  decimals: 18,
+  symbol: 'DAI',
+  occurrences: 1,
+  aggregators: [],
+  iconUrl: 'images/dai-logo.png',
+};
+
+const NATIVE_TOKEN = SWAPS_CHAINID_DEFAULT_TOKEN_MAP[CHAIN_IDS.MAINNET];
+
+const MOCK_SNAP: Snap = {
+  id: 'local:mock-snap' as SnapId,
+  origin: 'local:mock-snap' as SnapId,
+  version: '1.3.7' as SemVerVersion,
+  status: SnapStatus.Installing,
+  iconUrl: null,
+  initialPermissions: {},
+  sourceCode: 'sourceCode',
+  enabled: true,
+  blocked: false,
+  manifest: {
+    version: '1.3.7' as SemVerVersion,
+    initialPermissions: {},
+    manifestVersion: '0.1',
+    description: 'mock-description',
+    proposedName: 'mock-snap-name',
+    source: {
+      location: {
+        npm: {
+          registry: 'https://registry.npmjs.org',
+          filePath: 'dist/bundle.js',
+          packageName: 'local:mock-snap',
+        },
+      },
+      shasum: 'L1k+dT9Q+y3KfIqzaH09MpDZVPS9ZowEh9w01ZMTWMU=',
+      locales: ['en'],
+    },
+  },
+  versionHistory: [
+    {
+      date: 1680686075921,
+      origin: 'https://metamask.github.io',
+      version: '1.3.7' as SemVerVersion,
+    },
+  ],
+};
```

Comment:
Looks like we might want to consider making `@metamask/utils` a peerDep like our controllers. For now, this resolves the type errors:

```suggestion
const MOCK_SNAP = {
  id: 'local:mock-snap' as SnapId,
  origin: 'local:mock-snap' as SnapId,
  version: '1.3.7' as Snap['version'],
  status: SnapStatus.Installing,
  iconUrl: null,
  initialPermissions: {},
  sourceCode: 'sourceCode',
  enabled: true,
  blocked: false,
  manifest: {
    version: '1.3.7' as Snap['manifest']['version'],
    initialPermissions: {},
    manifestVersion: '0.1',
    description: 'mock-description',
    proposedName: 'mock-snap-name',
    source: {
      location: {
        npm: {
          registry: 'https://registry.npmjs.org',
          filePath: 'dist/bundle.js',
          packageName: 'local:mock-snap',
        },
      },
      shasum: 'L1k+dT9Q+y3KfIqzaH09MpDZVPS9ZowEh9w01ZMTWMU=',
      locales: ['en'],
    },
  },
  versionHistory: [
    {
      date: 1680686075921,
      origin: 'https://metamask.github.io',
      version: '1.3.7',
    },
  ],
} as const satisfies Snap & { origin: SnapId; iconUrl: string | null };
```

Reply:
Thanks for the suggestions, it is a good workaround, addressed in 3a05d054740a4b43e54a1bf0d455b4045cc60f61